src/effects/Stabilizer.h

```diff
@@ -0,0 +1,137 @@
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with OpenShot Library. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPENSHOT_STABILIZER_EFFECT_H
+#define OPENSHOT_STABILIZER_EFFECT_H
+
+#include "../EffectBase.h"
+
+#include <google/protobuf/util/time_util.h>
+
+#include <cmath>
+#include <stdio.h>
+#include <memory>
+#include "../Color.h"
+#include "../Json.h"
+#include "../KeyFrame.h"
+#include "stabilizedata.pb.h"
+
+using namespace std;
+using google::protobuf::util::TimeUtil;
+
+// Store the relative transformation parameters between consecutive frames
+struct EffectTransformParam
+{
+    EffectTransformParam() {}
+    EffectTransformParam(double _dx, double _dy, double _da) {
+        dx = _dx;
+        dy = _dy;
+        da = _da;
+    }
+
+    double dx;
+    double dy;
+    double da; // angle
+};
+
+// Stores the global camera trajectory for one frame
+struct EffectCamTrajectory
+{
+    EffectCamTrajectory() {}
+    EffectCamTrajectory(double _x, double _y, double _a) {
+        x = _x;
+        y = _y;
+        a = _a;
+    }
+
+    double x;
+    double y;
+    double a; // angle
+};
+
+
+namespace openshot
+{
+	
+	/**
+	 * @brief This class stabilizes video clip to remove undesired shaking and jitter.
+	 *
+	 * Adding stabilization is useful to increase video quality overall, since it removes 
+	 * from subtle to harsh unexpected camera movements.
+	 */
+	class Stabilizer : public EffectBase
+	{
+	private:
+		/// Init effect settings
+		void init_effect_details();
+		std::string protobuf_data_path;
+		Keyframe zoom;
+
+	public:
+	std::string teste;
+        std::map <size_t,EffectCamTrajectory> trajectoryData; // Save camera trajectory data
+    	std::map <size_t,EffectTransformParam> transformationData; // Save transormation data 
+
+		/// Blank constructor, useful when using Json to load the effect properties
+		Stabilizer(std::string clipTrackerDataPath);
+
+		/// Default constructor
+		Stabilizer();
+
+		/// @brief This method is required for all derived classes of EffectBase, and returns a
+		/// modified openshot::Frame object
+		///
+		/// The frame object is passed into this method, and a frame_number is passed in which
+		/// tells the effect which settings to use from its keyframes (starting at 1).
+		///
+		/// @returns The modified openshot::Frame object
+		/// @param frame The frame object that needs the effect applied to it
+		/// @param frame_number The frame number (starting at 1) of the effect on the timeline.
+		std::shared_ptr<Frame> GetFrame(std::shared_ptr<Frame> frame, int64_t frame_number) override;
+
+		std::shared_ptr<openshot::Frame> GetFrame(int64_t frame_number) override { return GetFrame(std::shared_ptr<Frame> (new Frame()), frame_number); }
```

Comment:
```suggestion
		std::shared_ptr<openshot::Frame> GetFrame(int64_t frame_number) override {
			return GetFrame(std::make_shared<openshot::Frame>(), frame_number);
		};
```

It's preferable to use `std::make_shared` when allocating new shared pointers. (And it's more concise, as a bonus.) 

Also, ultra-long lines make FeRD cry. :sob:  My rule of thumb: anytime GitHub starts horizontally scrolling, you done gone too far to the right.